RCTVideo.m

```diff
@@ -98,17 +96,37 @@ - (void)sendProgressUpdate
    if (video == nil || video.status != AVPlayerItemStatusReadyToPlay) {
      return;
    }
+    
+   CMTime playerDuration = [self playerItemDuration];
+   if (CMTIME_IS_INVALID(playerDuration)) {
+      return;
+   }
 
-  if (_prevProgressUpdateTime == nil ||
-     (([_prevProgressUpdateTime timeIntervalSinceNow] * -1000.0) >= _progressUpdateInterval)) {
-    [_eventDispatcher sendInputEventWithName:RNVideoEventProgress body:@{
-      @"currentTime": [NSNumber numberWithFloat:CMTimeGetSeconds(video.currentTime)],
-      @"playableDuration": [self calculatePlayableDuration],
-      @"target": self.reactTag
-    }];
+   CMTime currentTime = _player.currentTime;
+   const Float64 duration = CMTimeGetSeconds(playerDuration);
+   const Float64 currentTimeSecs = CMTimeGetSeconds(currentTime);
+   if( currentTimeSecs >= 0 && currentTimeSecs <= duration) {
+        [_eventDispatcher sendInputEventWithName:RNVideoEventProgress
+                                            body:@{
+                                                     @"currentTime": [NSNumber numberWithFloat:CMTimeGetSeconds(currentTime)],
+                                                     @"atValue": [NSNumber numberWithLongLong:currentTime.value],
+                                                     @"atTimescale": [NSNumber numberWithInt:currentTime.timescale],
```

Comment:
This is somewhat a general problem in `RCTVideo`. In our production code we can't use `currentTime`, because we need to be close to the `CMTime` model data. You just get nasty rounding errors if you get back and forth between `CMTimeGetSeconds` and `CMTime`. For that reason we just add two additional fields to every time we post to the JS side: `atValue` and `atTimescale`. Better names might be `currentTimeValue` and `currentTimescale`. But the important part here is that we have to preserve the `CMTime` data.